### PR TITLE
develop to main pull request

### PR DESCRIPTION
**Pull request by iszabo0150**

### CODE REVIEW

File: src/database/tables/samples_table_sql.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    instr_model     TEXT,
+    inlet           TEXT,
+    method_name     TEXT,
+    vial            REAL,
+    replicate       REAL,
+    injvol          REAL,
+    start_time      REAL,
+    end_time        REAL,
+    sampling_rate   REAL,
+    time            REAL,
```
Comment (chemplexity) [inline src/database/tables/samples_table_sql.txt]: lets remove the following items from the samples table: time, intensity, channel, and baseline

File: src/file/ExportNIST.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+% function to save a peak as a NIST formatted data structure
+
+% input will be sample info, peaks mass spectra (intensity(row, :)), and
+% channel (1,:)
+
+% function takes three inputs, 1: sample row, 2: peak mz array, 3: peak
+% intensity array
+
+% might be easier for it to just take in the data object here? 
+function data = ExportNIST(sampleRow, channelArray, intensityArray, timeArray)
```
Comment (chemplexity) [inline src/file/ExportNIST.m]: yes, lets have data as input so we can fill in the file_path, file_name, file_size

File: scripts/addChecksum.m [hunk 3]
```diff
@@ -0,0 +1,16 @@
+function data = addChecksum(data)
+
+% ------------------------------------------------------------------------
+% Method      : addChecksum(data)
+% Description : calls GetFileChesksum with an MD5 hash and adds the file
+% checksum as a field in each row of data
+% ------------------------------------------------------------------------
+
+for i=1: size(data, 1)
```
Comment (chemplexity) [inline scripts/addChecksum.m]: remove extra space before 'size'

File: scripts/getAverageSpectraOfTimeRange.m [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+% Description : averages the intensity spectra within the given time range
+% ------------------------------------------------------------------------
+
+if timeEnd <= timeStart
+
+    error('This is not a valid time range');
+
+end
+
+[startInd, endInd] = lookupTimeRangeIndex(timeArray, timeStart, timeEnd);
```
Comment (chemplexity) [inline scripts/getAverageSpectraOfTimeRange.m:18]: need to handle case where timeStart > all values in timeArray by throwing error message

File: scripts/preprocessData.m [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+% Method      : preprocessData
+% Description : crops the signal by time, applies the centroid algorithm 
+% to data.channel and data.intensity, and calculates and returns the
+% baseline
+% ------------------------------------------------------------------------
+
+baselineSmoothness = 1E7;
+baselineAsymmetry = 1E-4;
+
+for i=1: size(data, 1)
```
Comment (chemplexity) [inline scripts/preprocessData.m]: remove extra space before 'size'

File: src/database/CreateDatabase.m [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+end
+
+dbfile = p.Results.filename;
+conn = sqlite(dbfile, 'create');
+
+samplesTextArray = readlines('./src/database/tables/samples_table_sql.txt');
+samplesText = strjoin(samplesTextArray);
+sqlquery = samplesText;
+execute(conn, sqlquery);
+
```
Comment (chemplexity) [inline src/database/CreateDatabase.m:31]: lets add a display message saying DB is created and show its path/filename.db

File: src/database/tables/samples_table_sql.txt [hunk 7]
```diff
@@ -0,0 +1,35 @@
+create table samples
+(
+    sample_id       integer not null
+        constraint sample_id
+            primary key,
+    sample_name     TEXT,
+    file_name       TEXT    not null,
+    file_path       TEXT    not null,
+    sequence_index  integer not null,
```
Comment (chemplexity) [inline src/database/tables/samples_table_sql.txt]: remove not null from sequence_index

File: src/database/tables/peaks_table_sql.txt [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    peak_width   REAL    not null,
+    peak_height  REAL    not null,
+    area_Of      REAL,
+    error        REAL,
+    model        TEXT,
+    x_lim        REAL,
+    y_lim        REAL,
+    input_x      REAL,
+    input_y      REAL,
+    input_mode   TEXT,
```
Comment (chemplexity) [inline src/database/tables/peaks_table_sql.txt]: remove input_mode

File: tests/databaseUnitTest.m [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+% Test case 1
+preppedData = prepareDataSamples(data);
+assert(length(fieldnames(preppedData)) == 23);
+assert(isfield(preppedData, 'md5_checksum'));
+assert(length({preppedData.md5_checksum}) == length(preppedData));
+
+%% UpdateDatabaseSamples
+
+% Test case 1 (checks for closed connection)
+conn = sqlite('/Users/isabella/Desktop/gcms-pipeline/src/database/GCMS_Database.db');
```
Comment (chemplexity) [inline tests/databaseUnitTest.m]: use relative path

File: tests/unittestplaceholder.m [hunk 10]
```diff
@@ -0,0 +1,158 @@
+% Unit test placeholder
```
Comment (chemplexity) [inline tests/unittestplaceholder.m]: we can remove this file entirely, its ok to delete

**Review by chemplexity — CHANGES_REQUESTED** [review]

File: src/database/tables/samples_table_sql.txt (hunk 1 above)
Comment (iszabo0150) [reply src/database/tables/samples_table_sql.txt]: resolved

**Pull request merged**